terratorch/datasets/generic_pixel_wise_dataset.py

```diff
@@ -120,19 +121,38 @@ def __init__(
             )
         self.rgb_indices = [0, 1, 2] if rgb_indices is None else rgb_indices
 
-        self.dataset_bands = dataset_bands
-        self.output_bands = output_bands
+        bands_by_interval = (self._bands_defined_by_interval(bands_list=dataset_bands) and
+                             self._bands_defined_by_interval(bands_list=output_bands))
```

Comment:
The function _bands_defined_by_interval doing all these checks is not super pythonic. Could we not just have a function that goes through all the elements of `dataset_bands` and `output_bands` and, if they are an interval, expands them into the full list? (what `_generate_bands_intervals` does).

Then we avoid needing these checks because after this we are in a common format with no ranges.